core/arch/x86/encode.c

```diff
@@ -729,6 +738,10 @@ size_ok(decode_info_t *di /*prefixes field is IN/OUT; x86_mode is IN*/,
                 di->prefixes &= ~PREFIX_VEX_L;
                 return true;
             }
+            if (!X64_MODE(di) && (size_template == OPSZ_8_dimode16))
```

Comment:
This should be in resolve_var_x64_size(), not in this function.

Reply:
See above. Done.